File: src/main/java/de/thkoeln/inf/gpm/vgb/model/MedicalHistory.java
package de.thkoeln.inf.gpm.vgb.model;

import model.MedicalHistoryDao;
import util.DbUtil;

import java.util.List;

public class MedicalHistory {
    private Long id;

    public static MedicalHistory createOrUpdate(MedicalHistory medicalHistory) {
        return DbUtil.INSTANCE.runInTransaction(() -> MedicalHistoryDao.Companion.save(medicalHistory));
    }

    public static List<MedicalHistory> findAll() {
        return DbUtil.INSTANCE.runInTransaction(MedicalHistoryDao.Companion::findAll);
    }

    public static MedicalHistory findById(Long medicalHistoryId) {
        return DbUtil.INSTANCE.runInTransaction(() -> MedicalHistoryDao.Companion.get(medicalHistoryId).toMedicalHistory());
    }

    public static void delete(Long medicalHistoryId) {
        DbUtil.INSTANCE.runInTransaction(() -> MedicalHistoryDao.Companion.delete(medicalHistoryId));
    }

    public MedicalHistory(Long id) {
        this.id = id;
    }

    public MedicalHistory() {
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }
}

File: src/main/java/de/thkoeln/inf/gpm/vgb/model/MedicalHistory.kt
package de.thkoeln.inf.gpm.vgb.model

import org.jetbrains.exposed.dao.IntEntity
import org.jetbrains.exposed.dao.IntEntityClass
import org.jetbrains.exposed.dao.id.EntityID
import org.jetbrains.exposed.dao.id.IntIdTable
import org.jetbrains.exposed.sql.transactions.transaction

class MedicalHistory(val id: Int?)


class MedicalHistoryDao(id: EntityID<Int>) : IntEntity(id) {
    companion object : IntEntityClass<MedicalHistoryDao>(MedicalHistories) {
        /**
         * Update or create [MedicalHistory] in database
         */
        fun save(medicalHistory: MedicalHistory): MedicalHistory = transaction {
            if (medicalHistory.id != null) return@transaction medicalHistory
            new { }.toMedicalHistory()
        }
    }

    fun toMedicalHistory() = MedicalHistory(id.value)
}

object MedicalHistories : IntIdTable()